### ISSUE TITLE
[Feature Request]: Linux version

### ISSUE THREAD

**Issue by inmanturbo** [beyondcode/herd-community#329]
### Feature Description

I see windows is coming out. how about Linux Desktop? Should be fairly easy to support say, Ubuntu LTS, keeping in line with the FORGE server offerings.

### Is this feature valuable for other users as well and why?

Yes. There is a valet linux and it's awesome. Lots of linux users in the laravel world. Personally, I'm a daily driver. I get a new mac mini and mac book pro each time they come out, but I rarely even boot them up. I mostly just use them for testing safari occasionally. Was planning to use them for presentations and other work stuff but Ubuntu always ends up working just fine, and it's blazingly fast.

**Comment by joseph-d** [comment]
Totally agree that Herd isn't as badly needed on Linux as it is on Mac and Windows. Linux makes light work of a lot of what Mac / Windows users have to go through to get things working.

That said, the runtime features of Herd Pro such as dd(); nice log / mail capture and of course Xdebug that "just works" would be just as valuable to Linux users as anyone else. Yes, Linux users are more comfortable using the command line, but I still bought a Tinkerwell license because sometimes the command line just isn't enough to be completely productive.

Personally, I'd be first in line to buy a Herd Pro Linux license. But I completely understand if I'm the only one (or two) who would!

**Comment by inmanturbo** [comment]
> Totally agree that Herd isn't as badly needed on Linux as it is on Mac and Windows. Linux makes light work of a lot of what Mac / Windows users have to go through to get things working.
> 
> That said, the runtime features of Herd Pro such as dd(); nice log / mail capture and of course Xdebug that "just works" would be just as valuable to Linux users as anyone else. Yes, Linux users are more comfortable using the command line, but I still bought a Tinkerwell license because sometimes the command line just isn't enough to be completely productive.
> 
> Personally, I'd be first in line to buy a Herd Pro Linux license. But I completely understand if I'm the only one (or two) who would!

This. The pro features are what I was thinking about. The integrated xdbug, etc. NTH. I also have a tinkerwell license :) . I don't use it much but when I do it's very helpful.

**Comment by GregJohnStewart** [comment]
I'll bump this and say as a new-to-laravel developer on Linux, it is useful to have the same dev experience as the laracasts, to help follow along

**Comment by danielrosehill** [comment]
Just adding one more voice to say that even as a long-term Linux user who uses the terminal throughout the day, I appreciate any tools that make life easier (my philosophy: if a GUI can do what a CLI can, I'll take the visual option every day). I'm sure there are plenty of others who feel similarly.

**Comment by simpleittools** [comment]
I know this is considered closed, and I respect that. And you JUST launched the Windows version which I am sure has your team working very hard from a support and improvement standpoint. So 100% respect. I just want to share my voice on the subject as well.
How needed is it on Linux? Maybe not "as much." But the truth is, the free functions of Herd were already functionally solved by XAMPP, Homestead, Docker, etc. That doesn't mean that Herd isn't 9000 times better (seriously, I wouldn't even mention XAMPP to someone now that I have used Herd).
You can't replace the standard by being as good as, you have to be better.
That is what you have accomplished.

Are Linux users more comfortable with the command line and/or utilizing Docker? Sure. But that doesn't mean it is anywhere near as nice an experience as Herd.
And, as contrary as it may seem, I think Linux users are going to be more likely to buy Herd Pro than a Windows or Mac user. Why? 
1. Linux users are willing to pay, we just want to be able to control the things we pay for.
2 . Because we are WAY more likely to utilize the advanced features.

Value to BeyondCode:
1. Though there are only a few comments here, these few should show there is interest.
2. Let's face it: we Linux users may be few (compared to Windows users), but we are loud about the things we love. We are basically passionate advertisements when we love your software.

**Comment by jeffegiovani** [comment]
🤯

**Comment by HiltiHons** [comment]
+1

**Comment by farisbasha** [comment]
+1

**Comment by Jackson5551** [comment]
+1

**Comment by realmocaccino** [comment]
+1

**Comment by farisbasha** [comment]
As a Linux user, I’d love to see Herd available for Linux. While Linux has powerful native tools, having the simplicity and efficiency of Herd would make Laravel development even more seamless across platforms. Please consider the possibility of bringing this great tool to the Linux community.

**Comment by brensondevalfa** [comment]
I will say what nobody wants to say. Laravel's core business is Forge, Vapor, and now Cloud. It's perfect for Windows and Mac users who want 'it just works' solutions. Using Linux as a development system might encourage self-hosting, and they can't allow that kind of 'heresy'.

**Comment by inmanturbo** [comment]
> I will say what nobody wants to say. Laravel's core business is Forge, Vapor, and now Cloud. It's perfect for Windows and Mac users who want 'it just works' solutions. Using Linux as a development system might encourage self-hosting, and they can't allow that kind of 'heresy'.

I have several clients that I manage "self hosted" Laravel for. And we primarily use forge for this. I wouldn't say self hosting is bad for Laravel business. The more people self hosting, or the more people adopting in general, the larger the pool of people or orgs that may be interested in the commercial tools and offerings.

**Comment by mpociot** [comment]
https://x.com/marcelpociot/status/1849798830167896177?s=46&t=cUkCtO-4PC7LZLkfo1H0xA

**Comment by simpleittools** [comment]
> https://x.com/marcelpociot/status/1849798830167896177?s=46&t=cUkCtO-4PC7LZLkfo1H0xA

The day this is availble, getting Herd Pro.

**Comment by simpleittools** [comment]
> I will say what nobody wants to say. Laravel's core business is Forge, Vapor, and now Cloud. It's perfect for Windows and Mac users who want 'it just works' solutions. Using Linux as a development system might encourage self-hosting, and they can't allow that kind of 'heresy'.

Everyone is entitled to their opinion, and I respect yours.
My opinion differs.

Laravel makes money from many angles. Not just deployment (though it is a big part), but other tools like Nova and Spark. The fact that they will be offering a direct hosting solution (in Cloud, which I know is much more than a hosting solution) shows continual growth.
In my experience (I acknowledge a limited sample size of people I personally know, and cannot cite sources), Linux users don't use Linux because of the $0 price tag. It is used because it allows us to use our computers as we feel best. I have seen Linux users willing to spend MORE on things that provide them value.

**Comment by patrickodacre** [comment]
Adding my voice to the crowd. It would be great if this were available on Linux. There are PHP extensions not available on Windows that I want to use. I just purchased Pro, but I will have to go back to something like Homestead for those projects that require these extensions, and that's a shame.

**Comment by jdecode** [comment]
+1

**Comment by viniciuscoutinh0** [comment]
+1

**Comment by lloricode** [comment]
+1

**Comment by horneberh** [comment]
+1

**Comment by toiq** [comment]
+1

**Comment by dtufood-kihen** [comment]
+1 Linux support - it's plainly wrong to assume Linux users don't need Herd!

**Comment by malzariey** [comment]
+1

**Comment by frogcom** [comment]
+1

**Comment by SantanuDatta** [comment]
Recently hopped into fedora, man I miss this app, really hoping linux support comes soon, its been a while since the last update shown on twitter.

**Comment by IceWolf0110** [comment]
+1, we need herd for better development experience

**Comment by besrabasant** [comment]
+1

**Comment by jdecode** [comment]
I have found a sweet spot (and it actually works pretty well)

TL;DR

1. [Docker + Compose](https://docs.docker.com/guides/frameworks/laravel) : generic, multi-language containers
2. [DDEV](https://ddev.com) : Docker setup for PHP based solutions (generic to all frameworks)
3. 🔥 [Fadogen](https://fadogen.app) : DDEV setup for Laravel (specific, and good)
   1. **This is the closest thing to Herd and it is dope**

--------------------------


Herd - Makes things super-simple (yes, kinda Loaded-XAMPP)
[Herd Lite](https://php.new/install/linux/8.4) - This is the one that gets installed in Linux when we use the php.new installation instructions per Laravel docs


However, here's the stuff that I use

Docker with Docker-compose ( [official guide for Laravel](https://docs.docker.com/guides/frameworks/laravel))
 - Flexibility for switching versions of PHP/DB/Redis etc (knowing how to build custom Docker images is a big plus)
 - Works for all languages

[DDEV](https://ddev.com)
 - Docker based solution for managing PHP solutions (pretty awesome on its own)

[Fadogen](https://fadogen.app) - This is the beauty
 - It is a Laravel specific setup of DDEV
 - I've been playing around with it (after spending about 5 years with Docker+Compose) and I love it

**Comment by tibor-src** [comment]
@sschlein even a non-GUI version for Linux would be helpful, as parallel to MacOS the IDEs nowadays run in remote environments (VMs).

It would be ideal to have the same setup for Laravel development as on local machine and not having to maintain two different setups.

Herd Lite is a good start, though its just PHP :)

If a few more of the services and the CLI herd command could be added it would really help.

**Comment by SPPayne** [comment]
I concur on the position that Linux devs are normally more comfortable with managing their own environments. However, in smaller teams where the configuration files need to be useable between different operating systems, it is difficult to argue for maintaining a configuration file for a specific operating system that will need to be managed by one developer who has a particular preference. It's time and effort that cannot be justified; as a former manager I concur with that from my team's perspective. It's time and effort in my daily work I cannot justify, and it's not a hill I'm willing to die on over my job.

I've made the switch to Windows, and WSL has took the brunt of some of the pain, but if someone told me tomorrow that one of the other virtualisation tools works with the config files for Herd, I'd be reinstalling Linux yesterday.

It utterly frustrates me that this is the situation; it's almost like admitting that Herd is a "toolset for babies" because "real devs know how to virtualise, so why would we cater for them".

**Comment by sohailahmad07** [comment]
Any updates on this

**Comment by inmanturbo** [comment]
Unfortunately I'm sure this will probably not happen. Linux just being such a hard and unconstrained target. Not to mention all the reverse engineering that will likely happen.

**Comment by iqbalfarhan** [comment]
+1

**Comment by sligregni** [comment]
+1

**Comment by bendeckdavid** [comment]
+1

**Comment by Tor2r** [comment]
+1

**Comment by mzyxnuel** [comment]
+1

**Comment by countradooku** [comment]
+1

**Comment by emilianotisato** [comment]
+1

**Comment by alexramsey92** [comment]
Due to the increasing annoyances of Microsoft, Windows, their cloud and their AI increasingly embedding itself, I am moving my primary Desktop to ubuntu. I am also looking seriously into Mac, but I need to find $3,000 under a couch cushion to make that move. I am +1'ng this.

100% onboard to help develop this if it can be community developed.

**Comment by countradooku** [comment]
@alexramsey92 i want to do this. I had an attempt but it was not very high quality proof

**Comment by Tor2r** [comment]
@sschlein 😇

Any "news"....? ❤️

**Comment by geodro** [comment]
Check this out, it is Herd like and works on Linux 

https://geodro.github.io/lerd/
